lib/cyclonedx/base.rb

```diff
@@ -24,20 +25,32 @@ def build_metadata
     # Returns the 'components' object for a supported/unsupported scanner's report
     def build_components_object
       components = []
-      @scan_report.info[:dependencies].each do |dependency|
-        component = {
-          "bom-ref": "",
-          "type": DEFAULT_COMPONENT_TYPE,
-          "group": "",
-          "name": dependency[:name],
-          "version": "",
-          "purl": ""
-        }
-
-        # TODO: Add specific component parsing for individual scanners
-        components << component
+      info = @scan_report.to_h.fetch(:info)
+      info[:dependencies].each do |dependency|
+        components << parse_dependency(dependency)
       end
       components
     end
+
+    def parse_dependency(dependency)
+      {
+        "bom-ref": package_url(dependency),
+        "type": DEFAULT_DEP_COMPONENT_TYPE,
+        "group": "", # TODO: add group or domain name of the publisher
+        "name": dependency[:name],
+        "version": version_string(dependency),
+        "purl": package_url(dependency),
+        "properties": [
+          {
+            "key": "source",
+            "value": dependency[:source]
```

Comment:
Does `dependency[:source]` always refer to the source code repository? If it does, a `vcs` type [external reference](https://cyclonedx.org/use-cases/#external-references) would be better than using a property.

Reply:
Ah, I just noticed further down that this might not always be the case. If the source does refer to a repo it would be good to include it as an external reference.

Reply:
Yeah, I think in this case the `dependency[:source]` is different from the source code repository, but will definitely keep this in mind if/when we do want to include that information